proxy_agent/src/proxy/proxy_connection.rs

```diff
@@ -3,17 +3,50 @@
 
 //! This module contains the connection context struct for the proxy listener, and write proxy processing logs to local file.
 
-use crate::common::error::Error;
+use crate::common::error::{Error, HyperErrorType};
 use crate::common::result::Result;
 use crate::common::{constants, hyper_client};
 use crate::proxy::Claims;
 use crate::redirector::{self, AuditEntry};
 use crate::shared_state::proxy_server_wrapper::ProxyServerSharedState;
 use crate::shared_state::redirector_wrapper::RedirectorSharedState;
+use http_body_util::Full;
+use hyper::body::Bytes;
+use hyper::client::conn::http1;
+use hyper::Request;
 use proxy_agent_shared::logger_manager::{self, LoggerLevel};
 use std::net::{Ipv4Addr, SocketAddr};
 use std::path::PathBuf;
+use std::sync::Arc;
 use std::time::Instant;
+use tokio::sync::Mutex;
+
+pub type RequestBody = Full<Bytes>;
+struct Client {
+    sender: http1::SendRequest<RequestBody>,
+}
+
+impl Client {
+    async fn send_request(
+        &mut self,
+        req: Request<RequestBody>,
+    ) -> Result<hyper::Response<hyper::body::Incoming>> {
+        if self.sender.is_closed() {
+            return Err(Error::Hyper(HyperErrorType::CustomString(
```

Comment:
Need be more specific Error type name.

Reply:
When forwarding the response to client, return different http status code for this error type, mostly connection closed.